### DISCUSSION TITLE
Runs Mold on *BSD ?

### DISCUSSION THREAD

**Discussion by theuserbl** [rui314/mold#1302] [Q&A] [answered]
It runs on Linux, but currently not on macOS and Windows.
But runs it on *BSD (FreeBSD, NetBSD, OpenBSD, ...) ?

I wonder, that it is a lesser problem to add additional hardware-support (different CPUs)  and that it is a bigger problem to add support of different operating systems.
I understand, that adding linking of Windows programs is hard, because Windows have the PE-format and the object files can be different. gcc on Windows created different object files then VC++.
But Linux, *BSD and I think also macOS are all using the ELF-format. Where is there the difference for the linker for the different operating systems?
That the libraries which are linked are on different places and that the libraries are different can not be the reason. Or I am wrong?

**Comment by rui314** [comment, marked as answer]
I believe mold works out of the box on BSDs, but I haven't tried that myself.
